testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeIdentityColumnCompatibility.java

```diff
@@ -52,30 +56,269 @@ public void testIdentityColumn()
+        onDelta().executeQuery("CREATE TABLE default." + tableName +
+                "(data INT, UPPERCASE_IDENTITY BIGINT GENERATED ALWAYS AS IDENTITY)" +
+                "USING DELTA " +
+                "LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "'" +
+                "TBLPROPERTIES ('delta.columnMapping.mode'='name')");
+        try {
+            onDelta().executeQuery("INSERT INTO default." + tableName + " (data) VALUES (1), (2), (3)");
+
+            assertThat(onTrino().executeQuery("SELECT * FROM delta.default." + tableName))
+                    .containsOnly(row(1, 1), row(2, 2), row(3, 3));
+
+            // Verify identify column is recognized even when the name contains uppercase characters
+            assertQueryFailure(() -> onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES (4, 4)"))
+                    .hasMessageContaining("Writing to tables with identity columns is not supported");
+            assertQueryFailure(() -> onTrino().executeQuery("UPDATE delta.default." + tableName + " SET data = 3"))
+                    .hasMessageContaining("Writing to tables with identity columns is not supported");
+            assertQueryFailure(() -> onTrino().executeQuery("DELETE FROM delta.default." + tableName))
+                    .hasMessageContaining("Writing to tables with identity columns is not supported");
+            assertQueryFailure(() -> onTrino().executeQuery("MERGE INTO delta.default." + tableName + " t USING delta.default." + tableName + " s " +
+                    "ON (t.data = s.data) WHEN MATCHED THEN UPDATE SET data = 1"))
+                    .hasMessageContaining("Writing to tables with identity columns is not supported");
+
+            // Verify column operations preserves the identify column name and property
+            onTrino().executeQuery("ALTER TABLE delta.default." + tableName + " ADD COLUMN new_col integer");
+            onTrino().executeQuery("ALTER TABLE delta.default." + tableName + " RENAME COLUMN new_col TO renamed_col");
+            onTrino().executeQuery("COMMENT ON COLUMN delta.default." + tableName + ".uppercase_identity IS 'test comment'");
+            onTrino().executeQuery("ALTER TABLE delta.default." + tableName + " DROP COLUMN renamed_col");
+
+            assertThat((String) onDelta().executeQuery("SHOW CREATE TABLE default." + tableName).getOnlyValue())
+                    .contains("UPPERCASE_IDENTITY BIGINT GENERATED ALWAYS AS IDENTITY");
+
+            // Verify the connector preserves column identity property when renaming columns containing uppercase characters
+            onTrino().executeQuery("ALTER TABLE delta.default." + tableName + " RENAME COLUMN uppercase_identity TO renamed_identity");
+            assertThat((String) onDelta().executeQuery("SHOW CREATE TABLE default." + tableName).getOnlyValue())
+                    .contains("renamed_identity BIGINT GENERATED ALWAYS AS IDENTITY");
```

Comment:
There are some more fine-grained identity properties like delta.identity.start or delta.identity.step.
Let's have a test that these are not lost when we do metadata changes on a table, or when we write a checkpoint file.